tests/behat/features/bootstrap/SocialDrupalContext.php

```diff
@@ -450,4 +451,49 @@ public function iWaitForTheInstallerBatchJobToFinish() {
     $this->getSession()->wait(1800000, 'jQuery("#updateprogress").length === 0');
   }
 
+  /**
+   * Add likes to node at the start of a test with existing users as authors.
+   *
+   * Creates like provided in the form:
+   * | title    | bundle | user      |
+   * | My event | event  | Jane Doe  |
+   * | ...      | ...    | ...       |
+   *
+   * @Given likes node with defined author:
+   */
+  public function createLikesWithDefinedAuthor(TableNode $nodesTable): void {
+
+    $entity_manager = \Drupal::service('entity_type.manager');
+    /** @var \Drupal\votingapi\VoteTypeInterface $vote_type */
+    $vote_type = $entity_manager->getStorage('vote_type')->load('like');
+
+    /** @var \Drupal\votingapi\VoteStorageInterface $vote_storage */
+    $vote_storage = \Drupal::entityTypeManager()->getStorage('vote');
+
+    foreach ($nodesTable->getHash() as $nodeHash) {
+      if (!isset($nodeHash['author'])) {
+        throw new \Exception("User is not specified as author when using the 'like node with defined author:' step.");
+      }
+
+      $owner = user_load_by_name($nodeHash['author']);
+      if ($owner === FALSE) {
+        throw new \Exception(sprintf("User with username '%s' does not exist.", $nodeHash['author']));
+      }
+
+      $node_id = $this->getNodeIdFromTitle($nodeHash['bundle'], $nodeHash['title']);
+      if ($node_id === NULL) {
+        throw new \Exception("Node '%s' does not exist.", $nodeHash['title']);
+      }
+
+      /** @var \Drupal\votingapi\VoteInterface $vote */
+      $vote = $vote_storage->create(['type' => 'like']);
+      $vote->setVotedEntityId($node_id);
+      $vote->setVotedEntityType('node');
```

Comment:
@volodymyr-sydor It would be more flexible to not hardcode `node` here, but allow to define it in step. Then we can also use it for comments or posts in the future.

Reply:
@ribel 
yes, we could do it but then the method can become more complex:
  because we expect to get a node but instead we can get a post, so - looks like we won't be able easy validate input data. and then we needed to define a type of entity and only then load and so on.

It looks like the easiest way will be just to add for Post a separate method when it needed

Reply:
Ok, let's keep it like this for now.